src/main/java/frc/bionic/swerve/SwerveModule.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Team 4909, Bionics
+ * Billerica Memorial High School
+ *
+ * Copyright:
+ *   2021 Bionics
+ *
+ * License:
+ *   MIT: https://opensource.org/licenses/MIT
+ *   See the LICENSE file in the project's top-level directory for details.
+ */
+
+package frc.bionic.swerve;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+
+public class SwerveModule
+{
+  // Gear ratio of first two pairs of gears.
+  // Yaw does not depend on the third pair.
+  private double             GEAR_RATIO_12;
+
+  // Gear ratio of all three pairs of gears.
+  // Wheel speed depends on all three.
+  private double             GEAR_RATIO_123;
+
+  // Maximum yaw speed in RPM
+  private double            MAX_YAW_SPEED_RPM;
+
+  // Wheel diameter in meters
+  private double            WHEEL_DIAMETER_METERS;
+
+  // Desired state -- velocity of wheel in RPM, angle in degrees
+  // Angles are measured counter-clockwise, with zero being "robot forward"
+  public double             desiredWheelSpeedRPM;
+  public double             desiredYawDegrees;
+  
+  // Devices, Sensors, Actuators
+  MotorCANSparkMaxNeo       m_driveMotorA;
```

Comment:
To make this class motor agnostic should we use the interface name here instead of MotorCANSparkMaxNeo?